api/admin/controller.py

```diff
@@ -411,9 +416,61 @@ def complaints(self, identifier_type, identifier):
         
         return response
 
+    def roles(self):
+        """Return a mapping from MARC codes to contributor roles."""
+        # TODO: The admin interface only allows a subset of the roles
+        # listed in model.py since it uses the OPDS representation of
+        # the data, and some of the roles map to the same MARC code.
```

Comment:
@leonardr do you have any thoughts on this?

Reply:
In most cases this is fine because some other role means more or less the same thing (that's why they have the same MARC role). Just don't offer one or the other.

Use CONTRIBUTOR_ROLE for 'ctb' instead of COLLABORATOR_ROLE.
Use AFTERWORD_ROLE for 'aft' instead of COLOPHON_ROLE.
Use ASSOCIATED_ROLE for 'asn' instead of UNKNOWN_ROLE.
Use PRODUCER_ROLE for 'pro' instead of EXECUTIVE_PRODUCER_ROLE.

PRIMARY_AUTHOR_ROLE is the tricky one. We distinguish the primary author from other authors by putting them at the front of the list of contributors, rather than using a different MARC code. Whether changing the primary author means rearranging them in the user interface or selecting 'Primary Author' from a dropdown, it should be possible to convey the user's intention by reordering the contributor list in the OPDS entry sent back to the server.

Reply:
Okay, I could probably manage that by using PRIMARY_AUTHOR_ROLE for the first author that was submitted in the form. I think the UI will keep them in the order they were in the OPDS entry.

Reply:
I don't think it's a big deal right now, but that sounds like there's no way to change who the primary author is.

Reply:
The form will show them with the primary author first, and you can remove the authors and re-add them in the order you want. It's a little annoying since you can't just move them, but it works.